### DISCUSSION TITLE
Airflow FileSensor

### DISCUSSION THREAD

**Discussion by Larochenpp** [apache/airflow#25311] [Q&A] [answered]
Hello

I've been trying to find for a while the piece of code that allows me to trigger an action when a file is received in a given directory. For example launch a bash script upon receipt of the NFS file. But it doesn't work despite my many tries.
- First, it does not read in the directory that I define: /mnt/ben; but rather in path/airflow/dag, the default directory where DAG scripts are stored. I also added /mnt/ben in "admin -> connection" with "Conn type=fs"
- Two, the script does not read the most recent file, but the first one I created.
Yesterday I modified the "airflow.contrib.sensors.file_sensor" class to change to "airflow.sensors.filesystem" but I still have the same error.
My installed Airflow version is v2.2.4.

Below is my code and logs.

Thanks for your help

<img width="1722" alt="sensorLogs" src="https://user-images.githubusercontent.com/110045650/181040129-512fad6a-c2ce-458a-bc13-3a164e05c4ae.PNG">
<img width="1158" alt="code sensor demande" src="https://user-images.githubusercontent.com/110045650/181041868-7dca81b0-b660-47d9-893d-25e5f4b668c2.PNG">
.

**Comment by potiuk** [comment, marked as answer]
You need to create a connection and specify "path" on it. This is how filesensor gets the path it acts on.

Your fs_conn_id="/mnt/ben"  is looking for a connection named "/mnt/ben" and then looks for a "path" there. You can create connection via UI Admin-> Connection and as written in the documentation this connection should have "path" extra defined:

```
    Allows for interaction with an file server.

    Connection should have a name and a path specified under extra:

    example:
    Connection Id: fs_test
    Connection Type: File (path)
    Host, Schema, Login, Password, Port: empty
    Extra: {"path": "/tmp"}
```

**Reply by potiuk** [reply]
Than what happens the "path" from the connection is joined with the "filepath" you specified.  Think of it that connection (via path extra) defines where to look for a file, and filepath determines what file it shoud look for.

**Reply by Larochenpp** [reply]
hello potiuk

Thank you for your answer, it helped me. I was able to define the reception directory of the file.
Regarding the second part of my concern, do you have an idea please? Now the script always reads the first file only, not the most recent. The goal for me is to make sure that with each new file, the script can only read the most recent one and that this reading is automatic when the file is received. Please what parameter should I add?

Thanks

**Reply by potiuk** [reply]
You can write your own operator to do that. Copying an operator and modifying it to fit your need is a 1st class citizen in Airflow. I do not know by heart all the operators available, but look through the code and find out if maybe there is such a functionality. If not - implement it - and then - best contribute it back to the community (lile more than 2100 contributors of Airflow who mostly implemented and contributed back what they needed).

**Reply by Larochenpp** [reply]
Ah OK thanks. I don't have good Python development skills. I am a system administrator. I will ask one of my colleagues who has the skills to help me on this.

Thanks again
